bayes_opt/logger.py

```diff
@@ -1,82 +1,155 @@
+    """
+    Return the default logger.
+
+    Parameters
+    ----------
+    verbose : int
+        Verbosity level of the logger.
+
+    is_constrained : bool
+        Whether the underlying optimizer uses constraints (this requires
+        an additional column in the output).
+
+    Returns
+    -------
+    ScreenLogger
+        The default logger.
+
+    """
     return ScreenLogger(verbose=verbose, is_constrained=is_constrained)
 
 
 class ScreenLogger(_Tracker):
+    """Logger that outputs text, e.g. to log to a terminal.
+    
+    Parameters
+    ----------
+    verbose : int
+        Verbosity level of the logger.
+
+    is_constrained : bool
+        Whether the logger is associated with a constrained optimization
+        instance.
+    """
+
     _default_cell_size = 9
     _default_precision = 4
 
     def __init__(self, verbose=2, is_constrained=False):
         self._verbose = verbose
         self._is_constrained = is_constrained
         self._header_length = None
-        super(ScreenLogger, self).__init__()
+        super().__init__()
 
     @property
     def verbose(self):
+        """Return the verbosity level."""
         return self._verbose
 
     @verbose.setter
     def verbose(self, v):
+        """Set the verbosity level.
+
+        Parameters
+        ----------
+        v : int
+            New verbosity level of the logger.
+        """
         self._verbose = v
 
     @property
     def is_constrained(self):
+        """Return whether the logger is constrained."""
         return self._is_constrained
 
     def _format_number(self, x):
+        """Format a number.
+
+        Parameters
+        ----------
+        x : number
+            
+
+        Returns
+        -------
+        A stringified, formatted version of `x`.
+        """
         if isinstance(x, int):
-                s = "{x:<{s}}".format(
-                    x=x,
-                    s=self._default_cell_size,
-                )
+            s = f"{x:<{self._default_cell_size}}"
         else:
-            s = "{x:<{s}.{p}}".format(
-                x=x,
-                s=self._default_cell_size,
-                p=self._default_precision,
-            )
+            s = f"{x:<{self._default_cell_size}.{self._default_precision}}"
 
         if len(s) > self._default_cell_size:
             if "." in s:
                 return s[:self._default_cell_size]
-            else:
-                return s[:self._default_cell_size - 3] + "..."
+            return s[:self._default_cell_size - 3] + "..."
         return s
 
     def _format_bool(self, x):
+        """Format a boolean.
+
+        Parameters
+        ----------
+        x : boolean
+            
+
+        Returns
+        -------
+        A stringified, formatted version of `x`.
+        """
         if 5 > self._default_cell_size:
-            if x == True:
+            if x:
                 x_ = 'T'
-            elif x == False:
+            else:
```

Comment:
These two codes have different behavior, it seems that the old one was intently done with if / elif.

```
>>> x = "false"
>>> if x:
...    print("got 'true'!")
...
got 'true'!
```

In the previous code it would ignore anything that's not a boolean

Reply:
> it seems that the old one was intently done with if / elif

I think you're giving me too much credit here 🙃 

I do see how this could potentially print something wrong if called with the wrong argument type. What do you think would be the best way to handle this case?